src/Orleans/AssemblyLoader/AssemblyProcessor.cs

```diff
@@ -106,6 +108,47 @@ private static void OnAssemblyLoad(object sender, AssemblyLoadEventArgs args)
             ProcessAssembly(args.LoadedAssembly);
         }
 
+        private static IEnumerable<Assembly> FindAssembliesWithOrleansReference()
```

Comment:
Do we want to populate these through reflection rather than explicit configuration?  Reflection is automatic, which is nice, but seems more error prone, and harder to control.